components/provisioning/org.wso2.carbon.identity.provisioning.connector.scim/src/main/java/org/wso2/carbon/identity/provisioning/connector/scim/SCIMProvisioningConnector.java

```diff
@@ -155,6 +158,11 @@ private void updateUser(ProvisioningEntity userEntity) throws IdentityProvisioni
 
             ProvisioningClient scimProvsioningClient = new ProvisioningClient(scimProvider, user,
                     httpMethod, null);
+            if (provisioningOperation.equals(ProvisioningOperation.PUT)) {
```

Comment:
Since an enum is being compared we can use == operator which is null safe or change the order as ProvisioningOperation.PUT.equals(provisioningOperation)